src/transformations/RenameEntitiesTransformation.cpp

```diff
@@ -0,0 +1,107 @@
+#include "../../include/transformations/RenameEntitiesTransformation.h"
+
+using llvm::isa, llvm::cast;
+using std::unique_ptr, std::find, std::vector, std::string;
+using clang::FunctionDecl, clang::VarDecl, clang::DeclRefExpr;
+
+// ------------ RenameEntitiesVisitor ------------
+
+RenameEntitiesVisitor::RenameEntitiesVisitor(Rewriter * rewriter, const bool rename_func, const bool rename_var,
+                                             const int seed, const int max_tokens,
+                                             const int max_token_len, const bool test):
+        rewriter(rewriter), rename_func(rename_func), rename_var(rename_var), test(test) {
+    gen = new mt19937(seed);
+    token_len_generator = createUniformIntGenerator(max_token_len);
+    tokens_num_generator = createUniformIntGenerator(max_tokens);
+    char_generator = createUniformIntGenerator(num_lat_chars);
+}
+
+discrete_distribution<int> RenameEntitiesVisitor::createUniformIntGenerator(const int num_elements) {
+    vector<double> probabilities;
+    for (int i = 0; i < num_elements; ++i) {
+        // Here we create a discrete uniform distribution with probability
+        // equals to 100 / num_elements
+        probabilities.push_back(100 / num_elements);
+    }
+    return discrete_distribution<int> (probabilities.begin(), probabilities.end());
```

Comment:
Why we can't use simple [discrete uniform distribution](http://www.cplusplus.com/reference/random/uniform_int_distribution/) in range [1; num_elements]?